Update pyhomematic 0.1.54 + small fixes

danielperna84 opened the pull request:
## Description:
In pyhomematic:
- Set SABOTAGE helper for HmIP devices to channel 0
- Fix log-message for callbacks
- Add support for HmIP-FBL, HmIP-SWDM, HmIP-SMI55

In Home Assistant:
- Fix #19619 
- Ensure that the event we are processing is handeled by the correct entity. See this issue for further details: https://github.com/danielperna84/pyhomematic/issues/193
@pvizeli do you see anything wrong with this approach? So far it seems to work.

**Related issue (if applicable):** fixes #19619 

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

danielperna84 committed:
Update pyhomematic + small fix

danielperna84 committed:
Add casting for ILLUMINATION

pvizeli commented:
The question would be, what is wrong on event subscription function they evaluate the correct channels and subscribe it. Theoretical, he should only subscribe to the right channel that we don't need to look every callback call like now

danielperna84 commented:
I think it's shomehow related to this particular device having `STATE` for the actor __and__ also as a sensor.

danielperna84 commented:
You're right. I didn't notice that.
I think for the short term it would be best to leave out that fix from this PR and handle it separaely.

danielperna84 committed:
Revert suggested fix

pvizeli commented:
I think we can fix that there: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/homematic/__init__.py#L832-L836

My problem is, I can't remember why we do that. I think we should be fine always to use the channel or 0 and filter others out...

danielperna84 commented:
> My problem is, I can't remember why we do that. I think we should be fine always to use the channel or 0 and filter others out...

I don't remember either. How about instead of doing my first approach

```python
if int(device.split(':')[-1]) != int(self._channel):	
    return
```

we do this:

```python
eventchannel = int(device.split(':')[-1])
if eventchannel != int(self._channel) and eventchannel != 0:
    return
```

That should let events from the maintenance channel though, but block anything else that doesn't match the channel of the entity.
